kyuubi-server/src/main/scala/org/apache/kyuubi/engine/ProcBuilder.scala

```diff
@@ -190,6 +191,10 @@ trait ProcBuilder {
     if (logCaptureThread != null) {
       logCaptureThread.interrupt()
     }
+    if (!waitCompletion && process != null) {
+      info("Destroy the process, since waitCompletion is false.")
+      process.destroyForcibly()
+    }
   }
```

Comment:
Let's synchronized the method `close()`

Reply:
> Let's synchronized the method `close()` and reset `process` to `null`

Do we also need to reset `logCaptureThread` to `null`?

Reply:
> > Let's synchronized the method `close()` and reset `process` to `null`
> 
> Do we also need to reset `logCaptureThread` to `null`?

IMO, it's better to set `logCaptureThread` to null.